chat-plugins/blackjack.js

```diff
@@ -0,0 +1,399 @@
+			if (this.dealer.points > 21) {
+				for (let u in this.players) {
+					if (this.players[u].status === 'bust') continue;
+					winners.push(this.players[u].name);
+				}
+			} else if (this.dealer.points !== 21) {
+				for (let u in this.players) {
+					if (this.players[u].status === 'bust' || this.players[u].points <= this.dealer.points) continue;
+					winners.push(this.players[u].name);
+				}
+			} else if (this.dealer.points === 21) {
+				winners.push('The Dealer');
+			}
+
+			if (winners.length < 1) {
+				this.add(`<br />There are no winners this time.`, false, true);
+			} else {
+				this.add(Chat.html`<br />Winner${Chat.plural(winners.length)}: ${winners.join(', ')}`, false, true);
+			}
+		}
+
+		if (this.autostart) {
+			clearTimeout(this.autostart);
+			delete this.autostart;
+		}
+		this.state = 'ended';
+		this.ended = true;
+
+		this.destroy();
+	}
+	destroy() {
+		for (let i in this.players) {
+			this.players[i].destroy();
+		}
+		this.room.blackjack++;
+		delete this.room.game;
+	}
+	add(message, clean, add) {
+		if (clean) this.lastMessage = "";
+		this.room.send("|uhtmlchange|blackjack-" + this.id + "|");
+		if (!add) this.room.send("|uhtml|blackjack-" + this.id + "|<div class=\"infobox\">" + this.lastMessage + message + "</div>");
+		if (add) this.room.add("|uhtml|blackjack-" + this.id + "|<div class=\"infobox\">" + this.lastMessage + message + "</div>").update();
+		this.lastMessage = this.lastMessage + message;
+	}
+	hit(user) {
+		if (!this.started) return this.errorMessage(user, `Blackjack hasn't started yet.`);
+		if (!this.players[user]) return this.errorMessage(user, `You aren't a player in this game.`);
+		if (this.curUser !== user.userid) return this.errorMessage(user, `It's not your turn.`);
+
+		if (this.timer) {
+			clearTimeout(this.timer);
+			delete this.timer;
+		}
+
+		this.giveCard(user);
+	}
+	stand(user) {
+		let player = this.players[user];
+		if (!this.started) return this.errorMessage(user, `Blackjack hasn't started yet.`);
+		if (!player) return this.errorMessage(user, `You aren't a player in this game.`);
+		if (this.curUser !== user.userid) return this.errorMessage(user, `It's not your turn.`);
+		player.status = 'stand';
+		let cards = '';
+		for (let u in player.cards) cards += `[${player.cards[u]}]`;
+		this.add(Chat.html`<br /><b>${user.name}</b> stands with ${cards} (${player.points})`);
+
+		if (this.timer) {
+			clearTimeout(this.timer);
+			delete this.timer;
+		}
+
+		this.next();
+	}
+	giveCard(user) {
+		if (this.deck.length < 1) this.deck = Dex.shuffle(this.deck.slice(0));
+		let card = toId(this.deck[0]).toUpperCase();
+		let player = (user === 'dealer' ? this.dealer : this.players[user]);
+		if (!player) return; // this should never happen
+		player.cards.push(this.deck[0]);
+
+		this.deck.splice(0, 1);
+		let points = 0;
+
+		if (!isNaN(Number(card))) {
+			points += Number(card);
+		} else {
+			switch (card) {
+			case 'A':
+				if ((player.points + 11) <= 21) {
```

Comment:
I think you may have to recalculate the total for this instead.

A hand of [A, 9, K] would add up to 30 by your standard, since it does not acknowledge the Ace received at the start.